script.js

```diff
@@ -1 +1,283 @@
+      // newDeck.push(cardInfo);
+    }
+  }
+
+  return newDeck;
+};
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+const getRandomIndex = (max) => Math.floor(Math.random() * max);
+
+// Shuffle an array of cards
+const shuffleCards = (cards) => {
+  // Loop over the card deck array once
+  for (let currentIndex = 0; currentIndex < cards.length; currentIndex += 1) {
+    // Select a random index in the deck
+    const randomIndex = getRandomIndex(cards.length);
+    // Select the card that corresponds to randomIndex
+    const randomCard = cards[randomIndex];
+    // Select the card that corresponds to currentIndex
+    const currentCard = cards[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cards[currentIndex] = randomCard;
+    cards[randomIndex] = currentCard;
+  }
+  // Return the shuffled deck
+  return cards;
+};
+
+/* #######################
+## GLOBAL VARIABLES #####
+####################### */
+// boardSize has to be an even number
+const boardSize = 4;
+const board = [];
+let firstCard = null;
+let firstCardElement;
+let deck;
+
+/* ###########################
+## PLAYER ACTION CALLBACKS ##
+########################### */
+const squareClick = (messageBoard, cardElement, column, row) => {
+  console.log(cardElement);
+  console.log('FIRST CARD DOM ELEMENT', firstCard);
+  console.log('BOARD CLICKED CARD', board[column][row]);
+  const clickedCard = board[column][row];
+
+  // the user already clicked on this square
+  if (cardElement.innerText !== '') {
+    return;
+  }
+
+  // first turn
+  if (firstCard === null) {
+    console.log('first turn');
+    firstCard = clickedCard;
+    // turn this card over
+    cardElement.classList.add('card');
+    cardElement.innerHTML = `${firstCard.name}<br>${firstCard.suitSymbol}`;
+    messageBoard.innerText = 'Click on Another Card';
+    // hold onto this for later when it may not match
+    firstCardElement = cardElement;
+
+    // second turn
+  } else {
+    console.log('second turn');
+    // condition is met, first card matches second card
+    if (
+      clickedCard.name === firstCard.name
+    // && clickedCard.suit === firstCard.suit
+    ) {
+      console.log('match');
+      // display match message
+      // messageBoard.innerText = 'it\'s a match!';
+      messageBoard.innerHTML = '<img src="/Users/grahamlim/Documents/bootcamp/week-02/day-003/pre-class/match-game-bootcamp/doge_card.gif" />';
+      // apply css class and card's name and suit to cardElement so that it looks like the card has been turned over
+      cardElement.classList.add('card');
+      cardElement.innerHTML = `${clickedCard.name}<br>${clickedCard.suitSymbol}`;
+      card;
+    } else {
+      console.log('NOT a match');
+      messageBoard.innerText = 'no match, try again';
+      cardElement.innerHTML = `${clickedCard.name}<br>${clickedCard.suitSymbol}`;
+      cardElement.classList.add('card');
+
```

Comment:
when it is not a match, use a `canClick` variable to make the player unable to click on another card while waiting for the 2 cards to be cleared. If not, when the player clicks on a 3rd card while the 2 cards are still faced up, the 1st face up card will not disappear after 3 seconds because the 1st card is now the 3rd card due to line 123.